### PR TITLE
update docstrings

### PR DESCRIPTION
**Pull request by Yulya**
## Description:

**Related issue (if applicable):** fixes TATAU-<tatau issue number goes here>

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [ ] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated

If the code communicates with devices, web services, or third-party tools:
  - [ ] New or updated dependencies have been added to `setup.py.

### CODE REVIEW

File: tatau/nn/keras/model.py [hunk 1]
```diff
@@ -78,9 +112,27 @@ def generator(loader):
...
+        Start training process.
+        Collect metrics.
+
+        :param train_progress: Task Progress Callback
+        :param batch_size: batch size
+        :param chunk_dirs: list of paths to chunk dirs
+        :param current_iteration: iteration on whole train process
+        :param nb_epochs: number of epochs in current iteration
+        :param train_progress: instance of TrainProgress for reporting progress
+        :param dataset_meta: meta information of dataset
```
Comment (todofixthis) [inline tatau/nn/keras/model.py]: ❔ Could we include an explanation of what sort of meta information might be included in here, and where the meta information came from?

File: tatau/nn/keras/model.py [hunk 2]
```diff
@@ -78,9 +112,27 @@ def generator(loader):
...
+        :return: train history
```
Comment (todofixthis) [inline tatau/nn/keras/model.py]: ❔ Will a machine learning engineer who is not familiar with the Tatau codebase understand what a train history is?

File: tatau/nn/keras/model.py [hunk 3]
```diff
@@ -78,9 +112,27 @@ def generator(loader):
...
+        """
 
         self._net.compile(optimizer=self._optimizer, loss=self._loss_fn, metrics=self._metrics)
 
+        #  add if override this method
```
Comment (todofixthis) [inline tatau/nn/keras/model.py]: ❔ This is easy to overlook; could we put this information in the docstring for the method as well?  It might also be useful to include some example code in the docstring that a ML engineer can use as a starting point.

File: tatau/nn/keras/model.py [hunk 4]
```diff
@@ -107,6 +159,16 @@ def train(self, chunk_dirs: Iterable, batch_size: int, current_iteration: int,
         return train_history
 
     def eval(self, chunk_dirs: Iterable, batch_size: int, dataset_meta: dict = None):
+        """
+        Evaluate  model.
```
Comment (todofixthis) [inline tatau/nn/keras/model.py]: ❔ Could we make this a bit more explicit?  E.g., "Evaluate model accuracy" or similar.
✏️ Also, remove extra space char.

File: tatau/nn/keras/model.py [hunk 5]
```diff
@@ -78,9 +112,27 @@ def generator(loader):
 
     def train(self, chunk_dirs: Iterable, batch_size: int, current_iteration: int,
               nb_epochs: int, train_progress: tatau.TrainProgress, dataset_meta: dict = None):
+        """
+        Training process.
+
+        Configure model for training.
+        Initialize Progress Callback.
+        Start training process.
+        Collect metrics.
```
Comment (todofixthis) [inline tatau/nn/keras/model.py]: ❔ Could we format this as a [list](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks)?  It will make the Sphinx-generated documentation much easier to read.

File: tatau/nn/keras/model.py [hunk 6]
```diff
@@ -120,6 +182,8 @@ def init_distributed_backend(self, backend: str) -> None:
         """
         Init distributed backend.
 
+        Overriding is not recommended.
```
Comment (todofixthis) [inline tatau/nn/keras/model.py]: ❔ Could we include an explanation as to why it is not recommended to override this method?

File: tatau/nn/keras/model.py [hunk 7]
```diff
@@ -22,8 +22,8 @@ def __init__(self, net, optimizer, loss_fn, callbacks=(), metrics=('accuracy',))
 
         :param net: Keras model
         :param optimizer: Keras optimizer
-        :param loss_fn: loss function
-        :param callbacks: keras callbacks
+        :param loss_fn: loss function used in model training
+        :param callbacks: keras callbacks that are passed to training process
         :param metrics: model metrics
```
[the remark this answers is not in the document]
Comment (todofixthis) [reply tatau/nn/keras/model.py]: ❕ Overriding the initialiser seems to be a pretty common practice for creating Tatau models; could we also include some instructions for this?  E.g., indicate that the overridden method must invoke the superclass initialiser.

File: tatau/nn/keras/model.py [hunk 8]
```diff
@@ -107,6 +159,16 @@ def train(self, chunk_dirs: Iterable, batch_size: int, current_iteration: int,
...
+        """
+        Evaluate  model.
+
+        Start evaluation process by fitting appropriate generator
+
+        :param chunk_dirs: list of paths to chunk dirs
+        :param batch_size: batch size
+        :param dataset_meta: meta information of dataset
+        :return: tuple(loss, acc)
+        """
```
Comment (todofixthis) [inline tatau/nn/keras/model.py]: 💭 I might be going out on a limb here, but would it be correct to say that in most cases, the user can get the desired result by manipulating `loss_fn` and/or `metrics` in the initialiser, so that they don't have to override `eval`?  If so, we might include that info in this docstring, to help prevent ML engineers from going down an unnecessary rabbit hole.

File: tatau/nn/torch/model.py [hunk 9]
```diff
@@ -47,6 +54,19 @@ def __init__(self, net: Module, optimizer: Optimizer, criterion):
         self._optimizer = optimizer
 
     def get_data_loader(self, chunk_dirs: Iterable, batch_size, transform, target_transform, **kwargs):
+        """
+        Returns instance of PyTorch DataLoader.
+        Data Loader accepts NumpyChunkedDataset as a parameter
+        with args defined in this method.
+        collate_function method is passed as collate_function
```
Comment (todofixthis) [inline tatau/nn/torch/model.py]: ❔ I think this last line is meant to indicate that `collate_function` is passed via `kwargs`; is that right?

File: tatau/nn/torch/model.py [hunk 10]
```diff
@@ -60,6 +80,8 @@ def adjust_learning_rate(self, epoch: int):
         Adjust Learning Rate callback.
 
         :param epoch: current epoch
+
+        Overwrite in child class if you wan to adjust learning rate in optimizer
```
Comment (todofixthis) [inline tatau/nn/torch/model.py]: ✏️ `s/wan/want/`

File: tatau/nn/torch/model.py [hunk 11]
```diff
@@ -30,6 +30,13 @@ class Model(model.Model, ABC):
     weights_summarizer_class = 'tatau.nn.torch.summarizer.Mean'
 
     def __init__(self, net: Module, optimizer: Optimizer, criterion):
+        """
+        Constructor.
```
Comment (todofixthis) [inline tatau/nn/torch/model.py]: ❕ This method is pretty important, as it builds the PyTorch model; could we include more information here about what this method is used for, and when/how the user might need to override it?

File: tatau/nn/keras/model.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (todofixthis) [reply tatau/nn/keras/model.py]: ❔ Out of curiosity, why does the initialiser accept `net`, `optimizer`, `loss_fn`, etc. as parameters?  In all of the example models I've seen, the overridden method ignores these parameters and sets them explicitly.

File: tatau/nn/torch/model.py (hunk 11 above)
Comment (todofixthis) [reply tatau/nn/torch/model.py]: ❔ As with Keras, could we also ensure that every method has expected parameter types, example values, return types, and example code / instructions for overriding?
